src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -241,6 +241,28 @@ class PyStubCAS:
     def remove(self, digest: FileDigest | Digest) -> bool: ...
     def action_cache_len(self) -> int: ...
 
+# ------------------------------------------------------------------------------
+# Dependency inference
+# ------------------------------------------------------------------------------
+
+class NativeDependenciesRequest:
+    """A request to parse the dependencies of a file.
+
+    * The `digest` is expected to contain exactly one source file.
+    * Depending on the implementation, a json-serialized string `metadata`
+      can be passed. It will be supplied to the native parser, and
+      the string will be incorporated into the cache key.
+
+
+    Example:
+        result = await Get(NativeParsedPythonDependencies, NativeDependenciesRequest(input_digest, None)
+    """
+
+    def __init__(self, digest: Digest, metadata: str | None = None) -> None: ...
```

Comment:
Feels like we'd want the constructor to take in `metadata: Any = None` and have the serialization be done for the caller.

You can probably do this 2 ways:
- Move the type to Python and define `__init__`. In it use `json.dumps`.
- Keep this kinda as-is but more the serialization Rust-side. Feel free to make assumptions and document them.

I think the first option is easiest.
(Also, feel free to unconditionally serialize. `None` -> `"null"`, so meh...)

Reply:
Calling json.dumps from rust side of things is the smallest change, is that ok?